packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -142,11 +176,82 @@ export const treeNodeApi: TreeNodeApi = {
 
 /**
  * A collection of events that can be raised by a {@link TreeNode}.
+ *
+ * @privateRemarks
+ * TODO: add a way to subscribe to a specific field (for afterShallowChange and afterDeepChange).
+ * Probably have object node and map node specific APIs for this.
+ *
+ * TODO: ensure that subscription API for fields aligns with API for subscribing to the root.
+ *
+ * TODO: add more wider area (avoid needing tons of afterShallowChange registration) events for use-cases other than afterDeepChange.
+ * Some ideas:
+ *
+ * - afterDeepChange, but with some subtrees/fields/paths excluded
+ * - helper to batch several afterShallowChange calls to a afterDeepChange scope
+ * - parent change (ex: registration on the parent field for a specific index: maybe allow it for a range. Ex: node event takes optional field and optional index range?)
+ * - new content inserted into subtree. Either provide event for this and/or enough info to afterDeepChange to find and search the new sub-trees.
+ * Add separate (non event related) API to efficiently scan tree for given set of types (using low level cursor and schema based filtering)
+ * to allow efficiently searching for new content (and initial content) of a given type.
+ *
  * @public
  */
-export interface TreeNodeEvents {
+export interface TreeChangeEvents {
 	/**
-	 * Raised on a node right after a change is applied to one of its fields or the fields of a descendant node.
+	 * Emitted by a node when:
+	 *
+	 * - The content of one or more of the node's properties changes (i.e., the property now contains a new node, or
+	 * nothing if it previously contained a node).
+	 *
+	 * - For an array node, when an element is added, removed, or moved.
+	 *
+	 * - For a map node, when a key is added, updated, or removed.
+	 *
+	 * @remarks
+	 * In particular, this event is not raised when:
+	 *
+	 * - Properties of a child node change. Notably, updates to an array node or a map node (like adding or removing
+	 * elements/entries) will raise this event on the array/map node itself, but not on the node that contains the
+	 * array/map node as one of its properties.
+	 *
+	 * - The node is moved to a different location in the tree or removed from the tree.
+	 * In this case the event is raised on the _parent_ node, not the node itself.
+	 *
+	 * Also note that value nodes do not have properties (child nodes), so this event is never raised for them.
+	 *
+	 * For remote edits, this event is not guaranteed to occur in the same order or quantity that it did in
```

Comment:
```suggestion
```

Since, as I mentioned above, I don't think users of simple-tree will normally think of primitives as nodes (even though they are under the hood).

Reply:
I think Github tied a few of your suggestions to the wrong lines. Here you meant removing line 219, right?

Reply:
Correct. The other comment is wrong as well.